docs/notebooks/openai_gym_lunar_lander.pct.py

```diff
@@ -0,0 +1,244 @@
+    return total_reward
+
+
+# %% [markdown]
+# Original values for the various parameters take values between 0.05 and 1.0. We will set the search range for each parameter to be the same, from 0.0 to 1.2.
+
+# %%
+search_space = trieste.space.Box([0.0] * 12, [1.2] * 12)
+
+# %% [markdown]
+# Let's see what kind of reward we might get by just using random parameters from this search space. Usual reward values for the Lunar Lander environment are between -250 (terrible crash) to 250 (excellent landing).
+
+# %%
+for _ in range(10):
+    sample_w = search_space.sample(1).numpy()[0]
+    demo_heuristic_lander(env, sample_w, print_reward=True)
+
+# %% [markdown]
+# As you can see, most of the random sets of parameters result in a negative reward. So picking a value from this search space at random can result in a various unwanted behaviors. Here we show some examples of the landing not going according to plan. Each of these examples was created with a sample of the parameter values from the search space.
+#
+# **Warning:** all the videos in this notebook were pre-generated. Creating renders of OpenAI Gym environments requires various dependencies depending on software setups and operating systems, so we have chosen not to do it here in the interest of transferability of this notebook. For those interested in reproducing these videos, we have saved the input parameters and the code we used to generate them in the Trieste repository, in the folder next to this notebook. However because of the stochastic nature of the environment and the optimization described here, your results might differ slightly from those shown here.
+
+# %%
+import io
+import base64
+from IPython.display import HTML
+
+def load_video(filename):
+    video = io.open('./lunar_lander_videos/' + filename, 'r+b').read()
+    encoded = base64.b64encode(video)
+    return HTML(data='''
+            <video width="360" height="auto" alt="test" controls>
+                <source src="data:video/mp4;base64,{0}" type="video/mp4" />
+            </video>'''.format(encoded.decode('ascii'))
+    )
+
+
+# %% [markdown]
+# #### Crash
+
+# %%
+load_video("crash.mp4")
+
+# %% [markdown]
+# #### Timeout
+
+# %%
+load_video("timeout.mp4")
+
+# %% [markdown]
+# #### Flying out of bounds
+
+# %%
+load_video("out_of_bounds.mp4")
+
+# %% [markdown]
+# #### Slamming on the surface
+# This is a very common failure mode in this environment - going too fast and slamming on the surface.
+
+# %%
+load_video("slam.mp4")
+
+# %% [markdown]
+# Finally, we need a way to deal with stochasticity of the simulation. To keep our modelling simple, let's aim at finding a controller that achieves highest average reward over a fixed number of runs. This gives us a simple way of implementing the Trieste observer that isn't too sensitive to the randomness of a single simulation. Also note that we reverse the reward sign: this is because we seek the maximum reward, and Trieste approaches optimization as minimization problem.
```

Comment:
Sorry, still not happy with this sentence. How about:
"Since the reward is stochastic, our goal is to maximize its expectation (or rather here minimize minus its expectation since Trieste only deals with minimization problems). To decrease the observation noise, our observer returns the average of the reward over 10 runs."

Reply:
can't see the update?